pkg/resource/deploy/step_generator.go

```diff
@@ -178,6 +179,8 @@ func (sg *stepGenerator) GenerateSteps(
 	new := resource.NewState(goal.Type, urn, goal.Custom, false, "", inputs, nil, goal.Parent, goal.Protect, false,
 		goal.Dependencies, goal.InitErrors, goal.Provider, goal.PropertyDependencies, false,
 		goal.AdditionalSecretOutputs, goal.Aliases, &goal.CustomTimeouts)
+	// Mark the URN/resource as having been seen.
+	sg.urns[urn] = new
```

Comment:
I don't think it's safe to do this so late: this function can terminate before we reach this point if the processing of `ignoreChanges` fails. I think we should keep a second map from URN -> resource state instead, and leave the write to `sg.urns` where it originally was.